moped-editor/src/utils/mapHelpers.js

```diff
@@ -263,25 +313,23 @@ export const createProjectSelectLayerConfig = (
 
 /**
  * Create sources and layers for each source layer in the project's GeoJSON FeatureCollection
- * @param {object} selectedIds - Object containing selected ID array for each source layer
  * @param {object} geoJSON - A GeoJSON feature collection with project features
  * @return {JSX} Mapbox Source and Layer components for each source in the GeoJSON
  */
-export const createSummaryMapLayers = (selectedIds, geoJSON) => {
-  const geoJSONBySource = Object.keys(selectedIds).reduce(
-    (acc, sourceLayerName) => ({
-      ...acc,
-      [sourceLayerName]: {
-        ...geoJSON,
-        features: [
-          ...geoJSON.features.filter(
-            feature => feature.properties.sourceLayer === sourceLayerName
-          ),
-        ],
-      },
-    }),
-    {}
-  );
+export const createSummaryMapLayers = geoJSON => {
+  const geoJSONBySource = geoJSON.features.reduce((acc, feature) => {
+    const sourceLayerName = feature.properties.sourceLayer;
+
+    return acc[sourceLayerName]
+      ? {
+          ...acc,
+          [sourceLayerName]: {
+            ...geoJSON,
+            features: [...acc[sourceLayerName].features, feature],
+          },
+        }
+      : { ...acc, [sourceLayerName]: { ...geoJSON, features: [feature] } };
+  }, {});
```

Comment:
On fire 🔥  Wouldn't it have been easier to do something like this? 

```
return {
          ...acc,
          [sourceLayerName]: {
            ...geoJSON,
            features: [...(acc[sourceLayerName] ? sourceLayerName].features : []), feature],
          }
        }
```

Maybe it would be easier to read and less code duplication

Reply:
Yeah, this is a great idea! I'll update before merge. 🙏